src/dune_engine/process.ml

```diff
@@ -698,7 +698,14 @@ module Result = struct
       | No_capture -> ""
       | Read s -> s
       | File p ->
-        let contents = Stdune.Io.read_file p |> limit_output ~n:t.limit in
+        let all_contents =
+          match Stdune.Io.with_file_in ~f:Stdune.Io.read_all_unless_large p with
+          | Ok s -> s
+          | Error () ->
```

Comment:
I think this needs to at least print a warning to make the user aware that something is happening. Not sure if whatever `limit_output` is a good way to do it since it modifies the payload.

But also, depending on how this is being used (e.g. as input to another step - I am not sure this is possible here), this could be a correctness issue to truncate and thus an error might be the better outcome in this case. Silent truncation issues [can cause problems if undetected](https://www.bbc.com/news/technology-54423988), so maybe failing with a proper error message instead of a traceback would be the safer approach?

Reply:
Ah, I thought that this path was only used to display potentially truncated output, but it might be used elsewhere. I need to check.

Reply:
I think this would indeed corrupt the rules. You can confirm this with `%{read:..}` for example.